### PR TITLE
Rust enum variants from enumNameMappings config should be in PascalCase

### PR DESCRIPTION
**Pull request by maxnachlinger**
Fixes issue: https://github.com/OpenAPITools/openapi-generator/issues/18602 - Rust enum variants from enumNameMappings config should be in PascalCase

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh ./bin/configs/*.yaml
  ./bin/utils/export_docs_generators.sh
  ``` 
  (For Windows users, please run the script in [Git BASH](https://gitforwindows.org/))
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  IMPORTANT: Do **NOT** purge/delete any folders/files (e.g. tests) when regenerating the samples as manually written tests may be removed.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (upcoming 7.1.0 minor release - breaking changes with fallbacks), `8.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request. @frol (2017/07) @farcaller (2017/08) @richardwhiuk (2019/07) @paladinzh (2020/05) @jacob-pro (2022/10)

### CODE REVIEW

File: samples/client/petstore/rust/reqwest/petstore/src/models/order.rs [hunk 1]
```diff
@@ -49,7 +49,7 @@ pub enum Status {
     #[serde(rename = "approved")]
     Approved,
     #[serde(rename = "delivered")]
-    shipped,
+    Shipped,
```
Comment (maxnachlinger) [inline samples/client/petstore/rust/reqwest/petstore/src/models/order.rs:52]: Changes fixed this sample.

**Comment by wing328** [comment]
@maxnachlinger using the enum name mapping should allow users to have complete control of the enum name mapping. 

what about fixing it in the rust client generator (java class) instead?

**Comment by maxnachlinger** [comment]
@wing328 got it, works as designed.

**Comment by wing328** [comment]
If I'm not mistaken, the default naming convention for enum variable name is PascalCase, which should work for your use cases.

Let me know if my understanding is wrong or I miss some edge cases somehow.

**Pull request closed without merging**